src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/OnceInitializedOnceDisposedUnderLockAsync.cs

```diff
@@ -71,6 +108,30 @@ protected Task ExecuteUnderLockAsync(Func<CancellationToken, Task> action, Cance
             return ExecuteUnderLockCoreAsync(action, cancellationToken);
```

Comment:
Looks like you could merge the two `ExecuteUnderLockCoreAsync` methods, keeping only the generic `Task<T>` overload and passing a lambda here that produces an ignored return value. Would that introduce significant overhead?